File: command/command.py
# Command interface
class Command:
    def execute(self):
        pass


# Concrete command: turn the light on
class LightOnCommand:
    def __init__(self, light):
        self.light = light

    def execute(self):
        self.light.on()


# Concrete command: turn the light off
class LightOffCommand:
    def __init__(self, light):
        self.light = light

    def execute(self):
        self.light.off()


# Invoker
class Switch:
    def __init__(self, on_command, off_command):
        self.on_command = on_command
        self.off_command = off_command

    def flip_up(self):
        self.on_command.execute()

    def flip_down(self):
        self.off_command.execute()


# Receiver
class Light:
    def on(self):
        print('The light is on')

    def off(self):
        print('The light is off')


if __name__ == '__main__':
    light = Light()

    light_on_command = LightOnCommand(light)
    light_off_command = LightOffCommand(light)

    switch = Switch(light_on_command, light_off_command)

    switch.flip_up() # Output: The light is on
    switch.flip_down() # Output: The light is off

File: command/command.js
// Command
class Command {
    execute() {
        throw new Error('Method not implemented');
    }
}

// Concrete command: turn on the light
class LightOnCommand {
    constructor(light) {
        this.light = light;
    }

    execute() {
        this.light.on();
    }
}

// Concrete command: turn off the light
class LightOffCommand {
    constructor(light) {
        this.light = light;
    }

    execute() {
        this.light.off();
    }
}

// Invoker
class Switch {
    constructor(onCommand, offCommand) {
        this.onCommand = onCommand;
        this.offCommand = offCommand;
    }

    flip_up() {
        this.onCommand.execute();
    }

    flip_down() {
        this.offCommand.execute();
    }
}

// Receiver
class Light {
    on() {
        console.log('The light is on');
    }

    off() {
        console.log('The light is off');
    }
}

// Client code
const light = new Light();

const lightOnCommand = new LightOnCommand(light);
const lightOffCommand = new LightOffCommand(light);

const invokerSwitch = new Switch(lightOnCommand, lightOffCommand);

invokerSwitch.flip_up(); // Output: The light is on
invokerSwitch.flip_down(); // Output: The light is off
